pallets/parachain-system/src/lib.rs

```diff
@@ -836,919 +405,480 @@ decl_error! {
-			let mut taken = 0;
-			let mut result = Vec::new();
-			SENT_MESSAGES.with(|ms| ms.borrow_mut()
-				.retain(|m| {
-					let status = <Module::<Test> as GetChannelInfo>::get_channel_status(m.0);
-					let ready = matches!(status, ChannelStatus::Ready(..));
-					if ready && !ids.contains(&m.0) && taken < maximum_channels {
-						ids.insert(m.0);
-						taken += 1;
-						result.push(m.clone());
-						false
-					} else {
-						true
-					}
-				})
-			);
-			result
+	/// The message queue chain heads we have observed per each channel incoming channel.
+	///
+	/// This value is loaded before and saved after processing inbound downward messages carried
+	/// by the system inherent.
+	#[pallet::storage]
+	pub(super) type LastHrmpMqcHeads<T: Config> =
+		StorageValue<_, BTreeMap<ParaId, MessageQueueChain>, ValueQuery>;
+
+	#[pallet::storage]
+	pub(super) type PendingUpwardMessages<T: Config> =
+		StorageValue<_, Vec<UpwardMessage>, ValueQuery>;
+
+	/// The number of HRMP messages we observed in `on_initialize` and thus used that number for
+	/// announcing the weight of `on_initialize` and `on_finalize`.
+	#[pallet::storage]
+	pub(super) type AnnouncedHrmpMessagesPerCandidate<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	/// The weight we reserve at the beginning of the block for processing XCMP messages. This
+	/// overrides the amount set in the Config trait.
+	#[pallet::storage]
+	pub(super) type ReservedXcmpWeightOverride<T: Config> = StorageValue<_, Weight>;
+
+	/// The next authorized upgrade, if there is one.
+	#[pallet::storage]
+	pub(super) type AuthorizedUpgrade<T: Config> = StorageValue<_, T::Hash>;
+
+	#[pallet::inherent]
+	impl<T: Config> ProvideInherent for Pallet<T> {
+		type Call = Call<T>;
+		type Error = sp_inherents::MakeFatalError<()>;
+		const INHERENT_IDENTIFIER: InherentIdentifier =
+			cumulus_primitives_parachain_inherent::INHERENT_IDENTIFIER;
+
+		fn create_inherent(data: &InherentData) -> Option<Self::Call> {
+			let data: ParachainInherentData = data
+				.get_data(&Self::INHERENT_IDENTIFIER)
+				.ok()
+				.flatten()
+				.expect("validation function params are always injected into inherent data; qed");
+
+			Some(Call::set_validation_data(data))
 		}
-	}
 
-	impl DownwardMessageHandler for SaveIntoThreadLocal {
-		fn handle_downward_message(msg: InboundDownwardMessage) -> Weight {
-			HANDLED_DOWNWARD_MESSAGES.with(|m| {
-				m.borrow_mut().push(msg);
-			});
-			0
+		fn is_inherent(call: &Self::Call) -> bool {
+			matches!(call, Call::set_validation_data(_))
 		}
 	}
 
-	impl XcmpMessageHandler for SaveIntoThreadLocal {
-		fn handle_xcmp_messages<'a, I: Iterator<Item=(ParaId, RelayBlockNumber, &'a [u8])>>(
-			iter: I,
-			_max_weight: Weight,
-		) -> Weight {
-			HANDLED_XCMP_MESSAGES.with(|m| {
-				for (sender, sent_at, message) in iter {
-					m.borrow_mut().push((sender, sent_at, message.to_vec()));
+	impl<T: Config> GetChannelInfo for Pallet<T> {
```

Comment:
in general people tends to put in the pallet module only the items required by the pallet macro.
And they put other stuff outside of the macro.
Though I don't have strong opinion.